pilot/pkg/model/jwks_resolver.go

```diff
@@ -384,7 +401,7 @@ func (r *JwksResolver) refresher() {
 	for {
 		select {
 		case <-r.refreshTicker.C:
-			currentHasError := r.refresh()
+			currentHasError := r.refresh(networkFetchRetryCountOnRefreshFlow)
```

Comment:
if we are doing multiple retries here, then it seems like the refreshInterval modifications below will not be respected as expected anymore.

Also, we may be blocking here for a long time and not allow `jwksuriChannel` part to make progress

Reply:
We are already doing multiple retries for the refresh, we were not passing it in the method parameter. While making the call we are using these retries - https://github.com/istio/istio/blob/master/pilot/pkg/model/jwks_resolver.go#L454
I am not touching the existing flow instead I have created a new async process in the background.